docs/file-system-based-routing.md

```
@@ -0,0 +1,90 @@
+
+- Packages define URL segments: Each Python package within your application's root maps to a URL segment. Nesting directories creates nested routes. For example, the `my_app/users/` package will create routes under the `/users` URL path.
+- Special files mark application endpoints: `holm` looks for specific filenames within your application directory to discover pages, layouts, custom APIs, and compose your application.
+  - `page.py`: Creates a publicly accessible URL for a route segment.
+  - `layout.py`: Defines a shared UI that wraps a route segment and its children.
+  - `api.py`: Creates custom API endpoints for a route segment.
+
+## Routing conventions
+
+Let's use the following project structure as an example to explore the routing conventions in `holm`.
+
+```
```

markdownlint-cli2 (0.18.1)

Fenced code blocks should have a language specified

(MD040, fenced-code-language)